src/threechances.cpp

```diff
@@ -186,30 +192,123 @@ std::vector<Entity*> setInitPos(std::vector<Entity*> mv, EntityGrid* entityGrid,
 	return mv;
 }
 
+void log3dGrid(std::vector<std::vector<Coordinates>> coordSet) {
+	for (int i = 0; i < coordSet.size(); i++) {
+		printf("Start of Box %d\n", i);
+		for (int j = 0; j < coordSet[i].size(); j++) {
+			printf("X: %d, Y: %d\n", coordSet[i][j].x, coordSet[i][j].y);
+		}
+		printf("End of Box %d\n", i);
+	}
+}
+
+bool isCoordUsed(std::vector<Coordinates> coordSet, Coordinates target) {
+	for (size_t i = 0; i < coordSet.size(); i++) {
+		if (target == coordSet[i])
+			return true;
+	}
+
+	return false;
+}
+
 void ThreeChances::initializeEntities() {
+	// reset random seed
+	srand(time(NULL));
+
 	// Add to entity grid
 	entityGrid->addEntity(levelGrid->getStartTile(), PLAYER_ID);
 
 	std::vector<Entity*> mv = gameControl->getMonsterVec();
-
+	std::vector<std::vector<int>> gameGrid = levelGrid->getGrid();
+	
 	Entity *tempMonster;
 
-	for (size_t i = 0; i < DUCK_START_COORDS.size(); i++) {
-		tempMonster = new Duck;
-		tempMonster->initialize(this, TILE_SIZE, TILE_SIZE, DUCK_COLS, &duckTexture, DUCK_DATA);
-		mv = setInitPos(mv, entityGrid, tempMonster, entityGrid->getPlayerCoordinates(), DUCK_START_COORDS[i]);
-	}
+	if (gameMode == GAME_MODE::demo) {
+		int gridSize = gameGrid.size();
+
+		// initialize the 4 boxes
+		std::vector<std::vector<Coordinates>> coordSet;
+		for (int i = 0; i < 4; i++) {
+			std::vector<Coordinates> tempVec;
+			coordSet.push_back(tempVec);
+		}
 
-	for (size_t i = 0; i < GHOST_START_COORDS.size(); i++) {
-		tempMonster = new Ghost;
-		tempMonster->initialize(this, TILE_SIZE, TILE_SIZE, GHOST_COLS, &ghostTexture, GHOST_DATA);
-		mv = setInitPos(mv, entityGrid, tempMonster, entityGrid->getPlayerCoordinates(), GHOST_START_COORDS[i]);
+		// Add coordinates to that 4 boxes
+		for (int i = 0; i < gridSize; i++) {
+			for (int j = 0; j < gridSize; j++) {
+				
+				if (i < gridSize / 2 && j < gridSize / 2)
+					coordSet[0].push_back(Coordinates(j, i));
+				else if (i < gridSize / 2 && j >= gridSize / 2)
+					coordSet[1].push_back(Coordinates(j, i));
+				else if (i >= gridSize / 2 && j < gridSize / 2)
+					coordSet[2].push_back(Coordinates(j, i));
+				else 
+					coordSet[3].push_back(Coordinates(j, i));
+			}
+		}
+
+		// Spawn 5 monsters in each box
+		for (size_t i = 0; i < coordSet.size(); i++) {
+			int monsterCounter = 0;
+
+			do {
+				// try spawn random monster at random index
+				int randIndex = rand() % coordSet[i].size();
+				double randNo = ((double)rand()) / RAND_MAX;
+				Coordinates startCoord = coordSet[i][randIndex];
+
+				if (randNo < 0.5) {
+					// ghost 50%
+					tempMonster = new Ghost;
+					if (tempMonster->isValidSpawn(levelGrid, startCoord) && !entityGrid->isEntityOnGrid(startCoord)) {
```

Comment:
`!entityGrid->isEntityOnGrid(startCoord)` checks whether there is an entity on grid. If there isn't, it will spawn